### PR TITLE
Grant Tests Implementation

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

This PR makes the necessary changes to support grants in our adapter. All tests provided by dbt-core pass (with modifications). For grants to work, changes need to be made to the relevant "grant" macros. Also made a few miscellaneous changes including updating smoke_tests to not wipe out current profiles.yml file.

#### Macros
- **support_multiple_grantees_per_dcl_statement**: Dremio only supports one grantee per statement
- **get_show_grant_sql**: Gets data from privileges table
- **get_grant_sql** : Contains sql for granting a privilege
- **get_revoke_sql**: Contains sql for revoking a privilege
- **call_dcl_statements**: Dremio only supports calling one statement at a time

These macros are all called by the default apply_grants macro provided by dbt-core. For this reason, we need to make sure apply_grants is called in each of the materializations (table, view, seed, incremental). 

#### Tests
- **test_model_grants**: Tests grants for views and tables
- **test_incremental_grants**: Tests grants for incremental materializations
- **test_seed_grants**: Tests grants for seeds
- **test_snapshot_grants**: Tests grants for snapshots
- **test_invalid_grants**: Tests invalid grants by providing a invalid privilege and invalid user

#### Miscellaneous
- Added logic to keep current profiles.yml file in smoke_tests
- Removed the schema.sql macro, as those methods are overridden in impl.py
- Added logic to drop a table when drop_schema is called (only spaces allow the entire schema to be deleted)
- Added copyright headers to basic func tests

### Related Issue

https://github.com/dremio/dbt-dremio/issues/42

### Additional Reviewers

@ArgusLi @jlarue26

### CODE REVIEW

File: tests/functional/adapter/grants/test_incremental_grants.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+from tests.functional.adapter.grants.base_grants import BaseGrantsDremio
+from tests.functional.adapter.utils.test_utils import relation_from_name
+from dbt.tests.adapter.grants.test_incremental_grants import (
+    BaseIncrementalGrants,
+    user2_incremental_model_schema_yml,
+)
+
+
+class TestIncrementalGrantsDremio(BaseGrantsDremio, BaseIncrementalGrants):
+    # Define this here to use our modified version of relation_from_name
```
Comment (ravjotbrar) [inline tests/functional/adapter/grants/test_incremental_grants.py:31]: Explain why we don't need this for views

File: tests/functional/adapter/grants/test_invalid_grants.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# limitations under the License.
+
+
+from dbt.tests.adapter.grants.test_invalid_grants import BaseInvalidGrants
+from tests.functional.adapter.grants.base_grants import BaseGrantsDremio
+from tests.functional.adapter.utils.test_utils import relation_from_name
+from dbt.tests.util import get_connection
+
+# Currently we return an HTTP error
+# TODO: modify error handling to get message from Dremio console
```
Comment (ravjotbrar) [inline tests/functional/adapter/grants/test_invalid_grants.py:22]: Remove TODO and put actual link to an issue

File: tests/functional/adapter/utils/test_utils.py [hunk 3]
```diff
@@ -8,7 +22,7 @@
 
 logger = AdapterLogger("dremio")
 
-DATALAKE = "rav-test"
+DATALAKE = "dbt_test_source"
```
Comment (ravjotbrar) [inline tests/functional/adapter/utils/test_utils.py:25]: Add comment to add why we include underscores instead of dashes

Comment (ravjotbrar) [reply tests/functional/adapter/utils/test_utils.py:25]: Link to issue as well

File: tests/functional/adapter/utils/test_utils.py [hunk 4]
```diff
@@ -35,12 +49,15 @@ def relation_from_name(adapter, name: str, materialization=""):
         # if the relation is a view then use database
         if materialization == "view" or "view" in name:
```
Comment (ravjotbrar) [inline tests/functional/adapter/utils/test_utils.py:50]: put brackets around condition

Comment (ravjotbrar) [reply tests/functional/adapter/utils/test_utils.py:50]: black autoformats this to not include brackets

File: dbt/adapters/dremio/connections.py [hunk 5]
```diff
@@ -195,24 +195,20 @@ def drop_catalog(self, database, schema):
         credentials = connection.credentials
         api_parameters = connection.handle.get_parameters()
 
+        path_list = self._create_path_list(database, schema)
         if database != credentials.datalake:
```
Comment (ArgusLi) [inline dbt/adapters/dremio/connections.py:199]: What's the reason for this change?

File: dbt/adapters/dremio/connections.py [hunk 6]
```diff
@@ -195,24 +195,20 @@ def drop_catalog(self, database, schema):
...
+                    catalog_id=None,
+                    catalog_path=path_list,
+                    ssl_verify=False,
+                )
+            except DremioNotFoundException:
+                logger.debug("Catalog not found. Returning")
+                return
 
-        delete_catalog(api_parameters, catalog_info["id"], ssl_verify=False)
+            delete_catalog(api_parameters, catalog_info["id"], ssl_verify=False)
```
Comment (ArgusLi) [inline dbt/adapters/dremio/connections.py:211]: Why the change in indentation here?

**Review by ArgusLi — APPROVED** [review]
